baseclasses/decorators.py

```diff
@@ -0,0 +1,26 @@
+import functools
+import unittest
+
+
+def require_mpi(func):
+    try:
+        from mpi4py import MPI  # noqa
+
+        # this is a wrapper on func
+        # but we inject the MPI option in the signature
```

Comment:
Can you clarify how the option is injected here? And why we need to wrap `func` if we have `mpi4py`?

Reply:
Whoops this is an old comment, I tried this but decided not to do it, will update.

Reply:
Oh no I did remove this I think you were just looking at an old commit

Reply:
Oh yeah sry, I got tricked by the UI again -  and I was a bit confused bc I remembered you used a different approach in the end